src/Runner.Worker/Handlers/HandlerFactory.cs

```diff
@@ -55,7 +55,18 @@ public sealed class HandlerFactory : RunnerService, IHandlerFactory
             else if (data.ExecutionType == ActionExecutionType.NodeJS)
             {
                 handler = HostContext.CreateService<INodeScriptActionHandler>();
-                (handler as INodeScriptActionHandler).Data = data as NodeJSActionExecutionData;
+                var nodeData = data as NodeJSActionExecutionData;
+                
+                // With node12 EoL in 03/2022, we want to be able to uniformly upgrade JS actions to node16
+                if (executionContext.Global.Variables.GetBoolean("DistributedTask.ForceGithubJavascriptActionsToNode16") ?? false)
+                {             
+                    // The user can opt out of this behaviour by setting this variable to true
+                    if (StringUtil.ConvertToBoolean(Environment.GetEnvironmentVariable(Constants.Variables.Actions.AllowActionsUseUnsecureNodeVersion)))
```

Comment:
I think this only applies to self hosted runners. We may need to also check job env context so non-self hosted runners can opt out: https://github.com/actions/runner/blob/main/src/Runner.Worker/ActionCommandManager.cs#L167

Do we have access to that here?